### ISSUE TITLE
Reactivate skipped unit tests

### ISSUE THREAD

**Issue by jayaddison** [GourmandRecipeManager/gourmand#137]
### Steps to Reproduce
A few of the unit tests in the codebase [have been marked as skipped](https://github.com/GourmandRecipeManager/gourmand/blob/139ca0b72736a4d9de5d109d77a93447675dab3a/tests/test_web_importer.py#L8) temporarily.

### Expected Behavior
Most, or ideally all, tests that are currently skipped should run again.

### Current Behavior
Some of the unit tests are skipped.

### Possible Solution
Various changes may be required to fix this (one discovered issue so far is that [`backup_database` method](https://github.com/GourmandRecipeManager/gourmand/blob/139ca0b72736a4d9de5d109d77a93447675dab3a/src/gourmand/backends/db.py#L2113-L2114) is called with a `None` value for `filename` from some of the tests, resulting in a failure when `filename.with_name` is accessed -- perhaps constructing a database-object-without-backups-enabled object within the unit tests could help address that).

### Environment
* Operating System: Linux, Python3.10
* Version or commit id: 928b6962e8dd49fe011b55eddf34d92324c29885
* Installed as Flatpak or Python package?: N/A (run directly using `pytest` within a Python virtualenv)

**Comment by cydanil** [comment]
Yeah, the CI is... in a sorry state. So is the creation of appimages, both of which have been blocking new releases.  

The CI is super important, but I would prioritize the creation of appimage first, as then the manual release tests can be done, and a release comparable to the previous could be done.  

In the meantime, I welcome even the smallest fixes to the tests, though!

**Comment by jayaddison** [comment]
Is there a plan for refactoring-out the [`db.new_db` flag](https://github.com/GourmandRecipeManager/gourmand/blob/54d4047b4ced08679c9bc544b049fef9b26492f8/src/gourmand/backends/db.py#L190)?

**Comment by cydanil** [comment]
Not currently, but it can be safely removed.  

This flag was key to creating a new database on a new Gourmet installation.  
In Gourmand, a database with a couple of recipes is now provided by default.  
The flag is now never set to True and can safely be removed, along with the code for creating the database (although it should probably be isolated in its own script, should we want to create a skeleton later on).

**Comment by jayaddison** [comment]
> one discovered issue so far is that [backup_database method](https://github.com/GourmandRecipeManager/gourmand/blob/139ca0b72736a4d9de5d109d77a93447675dab3a/src/gourmand/backends/db.py#L2113-L2114) is called with a None value for filename from some of the tests, resulting in a failure when filename.with_name is accessed

...and today I discovered that @MrShark has already discovered and worked on that:

https://github.com/MrShark/gourmand/compare/fix_tests...MrShark:gourmand:testing

@MrShark do you have any plans to continue those fixes?  If not, I'd like to pick up from those commits, if that's ok with you.

**Comment by MrShark** [comment]
@jayaddison You are welcome to pick them up, I got squrelled away :-)

The most important patch is the two first I think: https://github.com/MrShark/gourmand/commit/b6b8d18854585bdc50cb3338f49bfeab67cf49af and https://github.com/GourmandRecipeManager/gourmand/commit/b5a5af4bca0fabedf678026b292ba52b69a5d6ce
As Cydanil points out you'll have to look into the removed show_message() call and reintroduce it some ware else.

The third patch is mostly me trying out pytest and I'm not sure of its value

**Comment by FriedrichFroebel** [comment]
I have been able to re-enable some of the disabled tests. In some cases, issues remain, for example with yield parsing.

**Comment by jayaddison** [comment]
Thank you @FriedrichFroebel - when I read through the code again today, I only found two tests that remain skipped, and I was able to get those to pass locally by removing the skip markers; so I'll open a pull request in a few moments to enable those too.
